src/account/mod.rs

```diff
@@ -242,6 +250,127 @@ pub struct Account {
     has_pending_changes: bool,
 }
 
+/// A thread guard over an account.
+#[derive(Debug, Clone)]
+pub struct AccountGuard(Arc<RwLock<Account>>);
```

Comment:
Guards are specifically named for their ability to provide access to an inner type while upholding some invariant. A better name for this type would be `SharedAccount`, `AccountHandle`, `AccountRef`, etc.

Reply:
this makes sense, i'll change it